### PR TITLE
Add support for Provider Account

### PR DESCRIPTION
**Pull request by jsmolar**

### CODE REVIEW

**Comment by jsmolar** [comment]
Review request: @pehala @jsmadis

File: threescale_api/resources.py [hunk 1]
```diff
@@ -1096,3 +1130,28 @@ def proxy(self) -> 'Proxy':
...
+        log.info("Changes the role of the user of the provider account to member")
+        return self.set_state(state='member', **kwargs)
+
+    def set_role_admin(self, **kwargs):
+        log.info("Changes the role of the provider account to admin")
+        return self.set_state(state='admin', **kwargs)
+
+    def suspend(self, **kwargs):
+        log.info("Changes the state of the user of the provider account to suspended")
+        return self.set_state(tate='suspend', **kwargs)
```
Comment (pehala) [inline threescale_api/resources.py:1149]: typo

File: threescale_api/resources.py [hunk 2]
```diff
@@ -1096,3 +1130,28 @@ def proxy(self) -> 'Proxy':
...
+
+    def unsuspend(self, **kwargs):
```
Comment (pehala) [inline threescale_api/resources.py:1151]: Are there any kwargs even possible?

File: threescale_api/resources.py [hunk 3]
```diff
@@ -1096,3 +1130,28 @@ def proxy(self) -> 'Proxy':
...
+        log.info("Revokes the suspension of a user of the provider account")
+        return self.set_state(state='unsuspend', **kwargs)
+
+    def activate(self, **kwargs):
+        log.info("Changes the state of the user of the provider account to active")
+        return self.set_state(state='activate', **kwargs)
```
Comment (pehala) [inline threescale_api/resources.py:1157]: Does this use the special endpoints that are used for this or will this just send the entire entity in update request?

File: threescale_api/resources.py (hunk 2 above)
Comment (jsmolar) [reply threescale_api/resources.py:1151]: No, I'll remove that

File: threescale_api/resources.py (hunk 3 above)
Comment (jsmolar) [reply threescale_api/resources.py:1157]: It uses its special endpoints:
image

Comment (pehala) [reply threescale_api/resources.py:1157]: Ah thanks!

**Comment by pehala** [comment]
LGTM

**Pull request merged**